app/commands/submission/analysis/process.rb

```diff
@@ -40,7 +40,9 @@ def handle_ops_error!
 
   def handle_completed!
     submission.analysis_completed!
+    submission.update(tags:)
     Solution::UpdateTags.(submission.solution)
+    Submission::LinkToMatchingApproach.defer(submission)
```

Comment:
I think we should do this inline, not via defer. My logic for this is that once `Process` finishes, we alert the user, and it would be nice to say at that stage that there's a relevant approach.